src/main/kotlin/org/rust/lang/core/types/ty/TyTypeParameter.kt

```diff
@@ -58,15 +61,22 @@ class TyTypeParameter private constructor(
     }
 
     companion object {
-        private val self = TyTypeParameter(Self, { emptyList() })
+        private val self = TyTypeParameter(Self, false, { emptyList() })
 
         fun self(): TyTypeParameter = self
 
-        fun self(trait: RsTraitOrImpl): TyTypeParameter =
-            TyTypeParameter(Self, { listOfNotNull(trait.implementedTrait) })
+        fun self(item: RsTraitOrImpl): TyTypeParameter {
+            val isSized = when (item) {
+                // TODO: support trait bounds
+                is RsTraitItem -> false
```

Comment:
I don't know if we really want to check trait bounds here because we get all bounds in `ImplLookup#rawFindImplsAndTraits` and it brings `Sized` trait in result set so related test passes because of this hack.

Reply:
All this code is a set of hacks and will be dismantled soon, so everything is OK =)